apps/nova/src/pages/ResetPassword.tsx

```diff
@@ -0,0 +1,139 @@
+import {
+  Button,
+  ButtonSize,
+  ButtonType,
+  Controller,
+  Field,
+  InputType,
+  Label,
+  toaster,
+  useFormatMessage,
+} from "@litespace/luna";
+import { useCallback, useEffect, useMemo } from "react";
+import { useForm } from "react-hook-form";
+import { useResetPassword } from "@litespace/headless/auth";
+import { Form, useNavigate, useSearchParams } from "react-router-dom";
+import { Route } from "@/types/routes";
+import { IUser } from "@litespace/types";
+import { useAppDispatch } from "@/redux/store";
+import { setUserProfile } from "@/redux/user/profile";
+import { resetTutorMeta } from "@/redux/user/tutor";
+import { resetUserRules } from "@/redux/user/schedule";
+
+interface IForm {
+  password: string;
+  newPassword: string;
+}
+
+const ResetPassword = () => {
+  const intl = useFormatMessage();
+  const navigate = useNavigate();
+  const params = useSearchParams();
+  const token = params[0].get("token");
+
+  useEffect(() => {
+    if (!token) return navigate(Route.Login);
+  }, []);
+
+  const { control, watch, formState, handleSubmit } = useForm<IForm>({
+    mode: "onSubmit",
+    defaultValues: {
+      password: "",
+      newPassword: "",
+    },
+  });
+  const dispatch = useAppDispatch();
+
+  const password = watch("password");
+  const newPassword = watch("newPassword");
+
+  const dispatchFn = (profile: IUser.ResetPasswordApiResponse) => {
+    dispatch(setUserProfile(profile));
+    dispatch(resetTutorMeta());
+    dispatch(resetUserRules());
+  };
+
+  const onSuccess = useCallback((profile: IUser.ResetPasswordApiResponse) => {
+    toaster.success({ title: intl("page.login.forget.password.compelete") });
+    dispatchFn(profile);
+    return navigate(Route.Root);
+  }, []);
+
+  const onError = useCallback((error: Error) => {
+    toaster.error({
+      title: intl("error.unexpected"),
+      description: error.message,
+    });
+  }, []);
+
+  const mutation = useResetPassword({ onSuccess, onError });
+
+  const onSubmit = useMemo(() => {
+    return handleSubmit(() => {
+      if (password !== newPassword) {
+        return toaster.error({ title: intl("error.passowrd.unequal") });
+      }
+      if (token && password.trim() !== "") {
+        mutation.mutate({ token, password });
+      }
```

Comment:
The `onSubmit` should not show error notifications. It should be handled before this. The verification should be done at the input level. 

```suggestion
      if (password !== newPassword || !token) return;
      mutation.mutate({ token, password });
```